src/Compilers/CSharp/Portable/Emitter/EditAndContinue/CSharpSymbolMatcher.cs

```diff
@@ -662,6 +677,9 @@ private static MethodSymbol SubstituteTypeParameters(MethodSymbol method)
             private bool AreNamedTypesEqual(NamedTypeSymbol type, NamedTypeSymbol other)
             {
                 Debug.Assert(s_nameComparer.Equals(type.Name, other.Name));
+                // TODO: Test with overloads (from PE base class?) that have modifiers.
+                Debug.Assert(!type.HasTypeArgumentsCustomModifiers);
```

Comment:
For some reason this class already makes this assumption for other types that can have modifiers. For example, AreArrayTypesEqual (line 626), etc. I am going to open a work item to confirm whether these are valid assumptions and add necessary handling if not.